sdk/src/main/java/io/horizen/utils/ListSerializer.java

```diff
@@ -41,7 +41,16 @@ public List<T> parse(Reader reader) {
         if(maxListLength > 0 && objectsCount > maxListLength)
             throw new IllegalArgumentException("Input data contains to many elements - " + objectsCount);
 
-        List<T> objectsList = new ArrayList<>(objectsCount);
+        //To avoid someone to create a malicious object which has a really big objectCount that can cause an Heap out of memory error
+        //we want to don't preallocate the ArrayList size except if this number of objects is less than 1000
```

Comment:
a couple of typos :-)
we want to don't -> we don't want
we decided to lowered -> we decided to lower